app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -511,42 +513,57 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope {
         }
     }
 
-    private fun externalAppLinkClicked(appLinkCommand: Command.HandleExternalAppLink) {
-        context?.let {
+    private fun openExternalDialog(intent: Intent, context: Context, fallbackUrl: String? = null, showFirstOption: Boolean = true) {
+        context.let {
             val pm = it.packageManager
-            val intent = appLinkCommand.appLink.intent
             val activities = pm.queryIntentActivities(intent, 0)
 
-            Timber.i("Found ${activities.size} that could consume ${appLinkCommand.appLink.url}")
-
             when (activities.size) {
                 0 -> {
-                    if (appLinkCommand.appLink.fallbackUrl != null) {
-                        webView?.loadUrl(appLinkCommand.appLink.fallbackUrl)
+                    if (fallbackUrl != null) {
+                        webView?.loadUrl(fallbackUrl)
                     } else {
                         showToast(R.string.unableToOpenLink)
                     }
                     return
                 }
-                1 -> {
-                    val activity = activities.first()
-                    val appTitle = activity.loadLabel(pm)
-                    Timber.i("Exactly one app available for intent: $appTitle")
-
-                    AlertDialog.Builder(it)
-                        .setTitle(R.string.launchingExternalApp)
-                        .setMessage(getString(R.string.confirmOpenExternalApp))
-                        .setPositiveButton(R.string.openExternalApp) { _, _ -> it.startActivity(intent) }
-                        .setNegativeButton(android.R.string.cancel) { dialog, _ -> dialog.dismiss() }
-                        .show()
-                }
                 else -> {
-                    val title = getString(R.string.openExternalApp)
-                    val intentChooser = Intent.createChooser(intent, title)
-                    it.startActivity(intentChooser)
+                    if (activities.size == 1 || showFirstOption) {
```

Comment:
The higher-level `when` is no longer useful here and just adds confusion. Can we factor it out?

Reply:
Done!